common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/model/codegen/BeanGenerator.java

```diff
@@ -0,0 +1,284 @@
+                .flatMapMaybe(this::transformToJavaFile)
+                .concatWith(Single.just(getJavaFileForBase()))
+                .doOnNext(javaFile -> javaFile.writeTo(getOutputFolder()))
+                .map(JavaFile::toJavaFileObject)
+                .map(javaFile -> getPath(javaFile, getOutputFolder()))
+                .toList()
+                .map(GenerationResult::of);
+
+        return generationResult;
+    }
+
+    public Maybe<JavaFile> transformToJavaFile(final AnyType object) {
+        return buildTypeSpec(object).map(typeSpec -> JavaFile.builder(getObjectPackage(getPackagePrefix(), object), typeSpec).build());
+
+    }
+
+    private JavaFile getJavaFileForBase() {
+        return JavaFile.builder(getBaseClassName().packageName(), getBaseClassTypeSpec()).build();
+    }
+
+    private Maybe<TypeSpec> buildTypeSpec(final AnyType type) {
+
+        final TypeSpec typeSpec;
+        if (getCustomTypeMapping().get(type.getName()) != null) {
+            return Maybe.empty();
+        } else if (type instanceof ObjectType) {
+            typeSpec = buildTypeSpecForObjectType(((ObjectType) type));
+        } else if (type instanceof StringType) {
+            typeSpec = buildTypeSpecForStringType((StringType) type);
+        } else {
+            return Maybe.error(new RuntimeException("unhandled type " + type));
+        }
+
+        return Maybe.just(typeSpec);
+    }
+
+    private TypeSpec buildTypeSpecForStringType(final StringType stringType) {
+        if (!stringType.getEnum().isEmpty()) {
+            TypeSpec.Builder enumBuilder = TypeSpec.enumBuilder(getClassName(getPackagePrefix(), stringType)).addModifiers(Modifier.PUBLIC);
+            stringType.getEnum()
+                    .stream()
+                    .map(StringInstance.class::cast)
+                    .map(StringInstance::getValue)
+                    .forEach(value-> addConstantToEnumConstant(enumBuilder,value));
+            enumBuilder.addJavadoc(getJavaDocProcessor().markDownToJavaDoc(stringType));
+            return enumBuilder.build();
+        } else {
+            TypeSpec typeSpec = TypeSpec.classBuilder(getClassName(getPackagePrefix(), stringType))
+                    .addModifiers(Modifier.PUBLIC)
+                    .superclass(getBaseClassName())
+                    .addJavadoc(getJavaDocProcessor().markDownToJavaDoc(stringType))
+                    .build();
+            return typeSpec;
+        }
+    }
+
+
+    private void addConstantToEnumConstant(TypeSpec.Builder enumBuilder, final String enumValue) {
+
+        TypeSpec anootationSpec = TypeSpec.anonymousClassBuilder("")
+                .addAnnotation(AnnotationSpec.builder(JsonProperty.class).addMember("value", "$S",enumValue).build())
+                .build();
+        enumBuilder.addEnumConstant(toEnumValueName(enumValue), anootationSpec);
+
+    }
+
+
+
+    private TypeSpec buildTypeSpecForObjectType(final ObjectType object) {
```

Comment:
Very nice method names, I really like them. But how about renaming the parameter to `objectType`.

Reply:
nice catch 👍